### PR TITLE
React intl filter by in provider details views

### PR DESCRIPTION
**Pull request by ddonahue007**
issue: https://issues.redhat.com/browse/COST-1372

https://user-images.githubusercontent.com/57504257/129579412-f9ab9051-bc52-4f0e-9a77-d280aa33fa8b.mov

### CODE REVIEW

File: src/pages/views/components/dataToolbar/dataToolbar.tsx [hunk 1]
```diff
@@ -417,9 +452,9 @@ export class DataToolbarBase extends React.Component<DataToolbarProps> {
   };
 
   private getDefaultCategoryOptions = (): ToolbarChipGroup[] => {
-    const { t } = this.props;
+    const intl = createIntlEnv();
 
-    return [{ name: t('filter_by.values.name'), key: 'name' }];
+    return [{ name: intl.formatMessage(messages.FilterByValuesName), key: 'name' }];
```
Comment (dlabrecq) [inline src/pages/views/components/dataToolbar/dataToolbar.tsx:457]: I don't believe `key` is being used in this message?

```
 FilterByValuesName: {
    defaultMessage: 'EN Name',
    description: 'Name',
    id: 'FilterByValuesName',
  },
```

File: src/pages/views/details/awsDetails/detailsToolbar.tsx [hunk 2]
```diff
@@ -85,21 +83,22 @@ export class DetailsToolbarBase extends React.Component<DetailsToolbarProps> {
...
       options.push({
-        name: t('filter_by.values.org_unit_id'),
+        name: intl.formatMessage(messages.FilterByValues, { value: 'org_unit_id' }),
         key: orgUnitIdKey,
       });
     }
     if (tagReport && tagReport.data && tagReport.data.length > 0) {
-      options.push({ name: t('filter_by.values.tag'), key: tagKey });
+      options.push({ name: intl.formatMessage(messages.FilterByValues, { value: 'tag' }), key: tagKey });
     }
```
Comment (dlabrecq) [inline src/pages/views/details/awsDetails/detailsToolbar.tsx:102]: I don't believe `key` is being used here?

File: src/pages/views/details/azureDetails/detailsToolbar.tsx [hunk 3]
```diff
@@ -76,19 +74,20 @@ export class DetailsToolbarBase extends React.Component<DetailsToolbarProps> {
...
-      { name: t('filter_by.values.service_name'), key: 'service_name' },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'service_name' }), key: 'service_name' },
       {
-        name: t('filter_by.values.resource_location'),
+        name: intl.formatMessage(messages.FilterByValues, { value: 'resource_location' }),
         key: 'resource_location',
       },
-      { name: t('filter_by.values.tag'), key: tagKey },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'tag' }), key: tagKey },
     ];
```
Comment (dlabrecq) [inline src/pages/views/details/azureDetails/detailsToolbar.tsx:91]: Again, the `key` isn't being used?

File: src/pages/views/details/gcpDetails/detailsToolbar.tsx [hunk 4]
```diff
@@ -76,17 +74,18 @@ export class DetailsToolbarBase extends React.Component<DetailsToolbarProps> {
...
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'account' }), key: 'account' },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'project' }), key: 'project' },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'service' }), key: 'service' },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'region' }), key: 'region' },
     ];
 
     if (tagReport && tagReport.data && tagReport.data.length > 0) {
-      options.push({ name: t('filter_by.values.tag'), key: tagKey });
+      options.push({ name: intl.formatMessage(messages.FilterByValues, { value: 'tag' }), key: tagKey });
     }
```
Comment (dlabrecq) [inline src/pages/views/details/gcpDetails/detailsToolbar.tsx:89]: I see how value is being used, but `key` appears to be unused.

File: src/pages/views/details/ibmDetails/detailsToolbar.tsx [hunk 5]
```diff
@@ -76,17 +74,18 @@ export class DetailsToolbarBase extends React.Component<DetailsToolbarProps> {
...
-      { name: t('filter_by.values.region'), key: 'region' },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'account' }), key: 'account' },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'project' }), key: 'project' },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'service' }), key: 'service' },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'region' }), key: 'region' },
     ];
 
     if (tagReport && tagReport.data && tagReport.data.length > 0) {
-      options.push({ name: t('filter_by.values.tag'), key: tagKey });
+      options.push({ name: intl.formatMessage(messages.FilterByValues, { value: 'tag' }), key: tagKey });
```
Comment (dlabrecq) [inline src/pages/views/details/ibmDetails/detailsToolbar.tsx:88]: Again, unused `key`.

File: src/pages/views/details/ocpDetails/detailsToolbar.tsx [hunk 6]
```diff
@@ -76,13 +74,14 @@ export class DetailsToolbarBase extends React.Component<DetailsToolbarProps> {
...
 
     const options = [
-      { name: t('filter_by.values.cluster'), key: 'cluster' },
-      { name: t('filter_by.values.node'), key: 'node' },
-      { name: t('filter_by.values.project'), key: 'project' },
-      { name: t('filter_by.values.tag'), key: tagKey },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'cluster' }), key: 'cluster' },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'node' }), key: 'node' },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'project' }), key: 'project' },
+      { name: intl.formatMessage(messages.FilterByValues, { value: 'tag' }), key: tagKey },
```
Comment (dlabrecq) [inline src/pages/views/details/ocpDetails/detailsToolbar.tsx:84]: unsed `key`

File: src/locales/messages.ts [hunk 7]
```diff
@@ -276,6 +276,77 @@ export default defineMessages({
...
+    defaultMessage: 'EN Choose value',
+    description: 'Choose value',
+    id: 'FilterByTagValuePlaceholder',
+  },
+  FilterByValues: {
+    defaultMessage:
+      '{value, select, account {EN Account} cluster {EN Cluster} name {EN Name} node {EN node} org_unit_id {EN Organizational unit} project {EN Project} region {EN Region} resource_location {EN Region} service {EN Service} service_name {EN Service_name} subscription_guid {EN Account} tag {EN Tag} other {}}',
+    description: 'Filter by values',
+    id: 'FilterByValues',
+  },
```
Comment (dlabrecq) [inline src/locales/messages.ts:344]: Curious if `defaultMessage` can span multiple lines for readability?

**Review by dlabrecq — CHANGES_REQUESTED** [review]

File: src/pages/views/components/dataToolbar/dataToolbar.tsx [hunk 8]
```diff
@@ -744,16 +781,17 @@ export class DataToolbarBase extends React.Component<DataToolbarProps> {
...
               value={tagKeyValueInput}
-              placeholder={t('filter_by.tag_value.input_placeholder')}
+              placeholder={intl.formatMessage(messages.FilterByTagValueInputPlaceholder)}
               onKeyDown={evt => this.onTagValueInput(evt)}
             />
             <Button
               isDisabled={isDisabled}
               variant={ButtonVariant.control}
-              aria-label={t('filter_by.tag_value.button_aria_label')}
+              // aria-label={t('filter_by.tag_value.button_aria_label')}
```
Comment (dlabrecq) [inline src/pages/views/components/dataToolbar/dataToolbar.tsx:793]: Can get rid of the comment

File: src/locales/messages.ts [hunk 9]
```diff
@@ -276,6 +276,77 @@ export default defineMessages({
     description: 'total cost is the sum of the infrastructure cost and supplementary cost',
     id: 'DashboardTotalCostTooltip',
   },
+  DetailsColumnManagementTitle: {
+    defaultMessage: 'EN Manage columns',
+    description: 'Manage columns',
+    id: 'DetailsColumnManagementTitle',
+  },
+  FilterByButtonAriaLabel: {
+    defaultMessage: 'EN Filter button for {filterKey} name',
```
[the remark this answers is not in the document]
Comment (ddonahue007) [reply src/locales/messages.ts:285]: I can change this to value, and yes it is being localized

**Review by dlabrecq — APPROVED** [review]

**Pull request merged**